components/core/src/compressor_frontend/OutputBuffer.hpp

```diff
@@ -0,0 +1,66 @@
+#ifndef COMPRESSOR_FRONTEND_OUTPUT_BUFFER_HPP
+#define COMPRESSOR_FRONTEND_OUTPUT_BUFFER_HPP
+
+// Project Headers
+#include "Buffer.hpp"
+#include "Token.hpp"
+
+/**
+ * A buffer containing the tokenized output of the parser.
+ * The active buffer contains all the tokens from the current log message.
+ * The first token contains the timestamp (if there is no timestamp the first token is invalid).
+ * For performance (runtime latency) it defaults to a static buffer and when more tokens are needed
+ * to be stored than the current capacity it switches to a dynamic buffer.
+ * Each time the capacity is exceeded a new dynamic buffer is added to the list of dynamic buffers.
+ */
+namespace compressor_frontend {
+
+    class OutputBuffer : public Buffer<Token> {
+    public:
+
+        /**
+         * Increment buffer pos, swaps to a dynamic buffer (or doubles its size) if needed
```

Comment:
* Should be in imperative form.
* Swapping to a dynamic buffer seems like an implementation detail (so it shouldn't be in the API header comment).
* It's not really clear to a reader what `pos` is. Perhaps this should be called `advance_to_next_token`?